File: src/opti_track/bebop_odom/scripts/receiveOdomTest.py
#!/usr/bin/env python

import rospy
from opti_msgs.msg import Odom

rigidBodyID = 1
neighborID1 = 2
neighborID2 = 3

nodeName = 'agent%s_alg'%rigidBodyID
topicName1 = 'agent%s_odom'%neighborID1
topicName2 = 'agent%s_odom'%neighborID2

odom1 = Odom()
odom2 = Odom()

###########################################
################ callback1 ################
###########################################
def callback1(msg):
    global odom1
    odom1 = msg
    rospy.loginfo("%s %s.%s",neighborID1, odom1.header.stamp.secs, odom1.header.stamp.nsecs)

###########################################
################ callback2 ################
###########################################
def callback2(msg):
    global odom2
    odom2 = msg
    rospy.loginfo("%s %s.%s",neighborID2, odom2.header.stamp.secs, odom2.header.stamp.nsecs) 
    
###########################################
################### sub ###################
###########################################
def sub():
    rospy.init_node(nodeName, anonymous=False)
    
    rospy.Subscriber(topicName1, Odom, callback1)
    rospy.Subscriber(topicName2, Odom, callback2)
    rospy.loginfo("OK")
    rospy.spin()

###########################################
##################  main ##################
###########################################
if __name__ == '__main__':
    try:
        sub()
    except rospy.ROSInterruptException:
        pass

File: src/opti_track/bebop_odom/src/receiveOdomTest.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include "opti_msgs/Odom.h"
#include <iostream>
#include <string>
#include <sstream>

#define rigidBodyID  1
#define neighborID1  2
#define neighborID2  3

opti_msgs::Odom odom1;
opti_msgs::Odom odom2;

/************************************************/
void Callback1(const opti_msgs::Odom::ConstPtr& msg)
{
  odom1 = *msg;
  ROS_INFO_STREAM(odom1.header.stamp); 
}
/************************************************/
void Callback2(const opti_msgs::Odom::ConstPtr& msg)
{
  odom2 = *msg;
  ROS_INFO_STREAM(odom2.header.stamp);
}
/************************************************/
int main(int argc, char **argv)
{
  std::stringstream nodeName;
  nodeName<<"agent"<<rigidBodyID<<"_alg";
  
  std::stringstream topicName1;
  topicName1<<"agent"<<neighborID1<<"_odom";
  ROS_INFO_STREAM("agent"<<neighborID1<<"_odom");
  //std::stringstream topicName2;
  //topicName2<<"agent"<<neighborID2<<"_odom";
  std::string topicName2;
  topicName2 = std::string("agent")+std::to_string(neighborID2)+std::string("_odom");
  
  ros::init(argc, argv, nodeName.str());
  ros::NodeHandle n;
  ros::Subscriber sub1 = n.subscribe(topicName1.str(), 100, Callback1);
  ros::Subscriber sub2 = n.subscribe(topicName2, 100, Callback2);
  
  
  
  ros::spin();

  return 0;
}
